tests/test_salesforce_switch_rep_method_ft_incrmntl.py

```diff
@@ -0,0 +1,96 @@
+
+        # Select only the expected streams tables
+        expected_streams = self.expected_sync_streams()
+        catalog_entries = [ce for ce in found_catalogs if ce['tap_stream_id'] in expected_streams]
+        self.select_all_streams_and_fields(conn_id, catalog_entries)
+        streams_replication_methods = {stream: self.FULL_TABLE
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # Run a sync job using orchestrator
+        fulltbl_sync_record_count = self.run_and_verify_sync_mode(conn_id)
+        fulltbl_sync_records = runner.get_records_from_target_output()
+
+        fulltbl_sync_bookmarks = menagerie.get_state(conn_id)
+
+        #Switch the replication method from full table to Incremental
+        streams_replication_methods = {stream: self.INCREMENTAL
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # SYNC 2
+        incrmntl_sync_record_count = self.run_and_verify_sync_mode(conn_id)
+        incrmntl_sync_records = runner.get_records_from_target_output()
+        incrmntl_sync_bookmarks = menagerie.get_state(conn_id)
+
+        # Test by stream
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # record counts
+                fulltbl_sync_count = fulltbl_sync_record_count.get(stream, 0)
+                incrmntl_sync_count = incrmntl_sync_record_count.get(stream, 0)
+                replication_key = list(replication_keys[stream])[0]
+
+                # Verify at least 1 record was replicated in the Incrmental sync
+                self.assertGreater(incrmntl_sync_count, 0,
+                                   msg="We are not fully testing bookmarking for {}".format(stream))
+                # data from record messages
+                primary_key = list(primary_keys[stream])[0]
+                fulltbl_sync_messages = [record['data'] for record in
+                                       fulltbl_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                fulltbl_primary_keys = {message[primary_key] for message in fulltbl_sync_messages}
+                incrmntl_sync_messages = [record['data'] for record in
+                                        incrmntl_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+                incrmntl_primary_keys = {message[primary_key] for message in incrmntl_sync_messages}
+
+                #Verify all records are synced in the second sync
+                self.assertTrue(fulltbl_primary_keys.issubset(incrmntl_primary_keys))
+
+                #verify that the last message is not a activateversion message for incremental sync
```

Comment:
What was the final decision on the leading 'action'?  I'd like to also verify that the leading 'action' is either activate_version or upsert as expected.

Reply:
This would depend on the ticket [TDL-24467](https://jira.talendforge.org/browse/TDL-24467) Will add  a comment for time being  to add/modify assertion based on the ticket implementation